core/src/main/scala/org/apache/spark/rpc/akka/AkkaRpcEnv.scala

```diff
@@ -305,7 +310,7 @@ private[akka] class AkkaRpcEndpointRef(
         }
       case AkkaFailure(e) =>
         Future.failed(e)
-    }.mapTo[T]
+    }(AkkaRpcEnv.askExecutionContext).mapTo[T]
```

Comment:
@pwendell `AkkaRpcEnv.askExecutionContext` is only used for `Future.flatMap`. And the logic in `flatMap` never blocks. So it's safe to just run it in the caller's thread.